### DISCUSSION TITLE
Is it possible to use pgx.RowToStructByName with QueryRow?

### DISCUSSION THREAD

**Discussion by stasvo** [jackc/pgx#1493] [Q&A] [answered]
Is it possible to use pgx.RowToStructByName with QueryRow for case when need to get only one row?
Or only way to use pgx.RowToStructByName is use together with Query and pgx.CollectRows even if I need to request only single row?

**Comment by jackc** [comment, marked as answer]
You can use `Query` and `pgx.CollectOneRow`.

**Reply by stasvo** [reply]
Thanks

**Reply by talksik** [reply]
this devex is a little odd to me. is there any systemic reason?
